src/rgw/motr/gc/gc.cc

```diff
@@ -145,33 +150,58 @@ bool MotrGC::going_down() {
   return down_flag;
 }
 
-int MotrGC::dequeue(const DoutPrefixProvider* dpp, std::string iname, motr_gc_obj_info obj)
-{
+int MotrGC::enqueue(motr_gc_obj_info obj) {
+  int rc = 0;
+  // create keys 0_{obj.tag} & 1_{obj.time + min_wait_time}
+  std::string key1 = obj_tag_prefix + obj.tag;
+  std::string key2 = obj_exp_time_prefix +
+                     std::to_string(obj.time + cct->_conf->rgw_gc_obj_min_wait);
+
+  bufferlist bl;
+  obj.encode(bl);
+  // push {0_ObjTag: motr_gc_obj_info} to the gc queue.📥
+  rc = store->do_idx_op_by_name(index_names[enqueue_index],
+                                M0_IC_PUT, key1, bl);
+  if (rc < 0)
+    return rc;
+  
+  // push {1_ExpiryTime: motr_gc_obj_info} to the gc queue.📥
+  rc = store->do_idx_op_by_name(index_names[enqueue_index],
+                                M0_IC_PUT, key2, bl);
+  if (rc < 0)
```

Comment:
here remove the added 0_ObjTag entry from index before returning error

Reply:
addressed.